internal/lint/base_visitor.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package lint
+
+import (
+	"text/scanner"
+
+	"github.com/emicklei/proto"
+	"github.com/uber/prototool/internal/text"
+)
+
+type baseVisitor struct{}
```

Comment:
Seeing as this is only used internally in the `newBaseAddVisitor`, it might be worth collapsing these types into one:

```
type baseVisitor struct {
  addFailure func(*text.Failure)
}
```

Reply:
Happy to do it, I’m personally a huge fan of composition but up to you

Reply:
Yea, the composition isn't bad. @AlexAPB what are your thoughts here? I think it might be a little simpler to only have one struct, but it's a minor detail.

Reply:
Either work for me, but I guess this brings up a different question:
@peter-edge did you have any ideas of adding different types of linters, other than `add`s? I feel like the current behaviour is the only behaviour we'd want, no?

Reply:
Yea I did, I didn't end up doing so. We can work to just make it add, but up to you - since the abstraction is already there, might be nice to leave it, but also might be nice to simplify.

Reply:
I'm of the mindset for simplifying if possible